src/nrniv/nrndae.cpp

```diff
@@ -75,31 +76,31 @@ void nrndae_init() {
 void nrndae_rhs(NrnThread* _nt) {
     update_sp13_mat_based_on_actual_d(_nt);
     update_sp13_rhs_based_on_actual_rhs(_nt);
-    for (NrnDAEPtrListIterator m = nrndae_list.begin(); m != nrndae_list.end(); m++) {
-        (*m)->rhs();
+    for (auto&& item: nrndae_list) {
```

Comment:
Maybe these could be `auto*` to make it a little bit clearer we're just dealing with some pointer.